### ISSUE TITLE
Crashes reported on iOS Prod build 

### ISSUE THREAD

**Issue by Gauravprakash** [getsentry/sentry-cocoa#2993]

**Comment by kahest** [comment]
Hi @Gauravprakash - thank you for reporting, we are investigating. Do you by any chance have the full stacktraces for those issues and could you share them with us?

**Comment by krystofwoldrich** [comment]
I've moved this to `sentry-cocoa` because the RN code snippet showed `enabledNative: false`. Definitely, a whole stack trace would be helpful.

**Comment by philipphofmann** [comment]
I think most of the issues should be fixed with https://github.com/getsentry/sentry-cocoa/pull/2995.

**Comment by armcknight** [comment]
> I think most of the issues should be fixed with #2995.

I think that's going to be a relatively small proportion of the overall amount of crashes. The screenshot shows a total of 210 crashes, but #2995 only addresses the ones from `SentryProfiler.processBacktrace`, which is only 5% of the total. We would really need a stacktrace to see what's up with `+ [SentryProfiler isRunning]`. I'm not sure what's up with the most prevalent crash in the serializer. Stack traces will definitely be helpful.

**Comment by brustolin** [comment]
I also would like to see the stacktrace for `+[SentrySerialization dataWithJSONObject:]`

@Gauravprakash You can also [send me](mailto:dhiogo.brustolin@sentry.io) your org name so we could better investigate.

**Comment by Gauravprakash** [comment]
please check attached stacktrace, I have extracted it from 2 major crash for now 

[Archive 2.zip](https://github.com/getsentry/sentry-cocoa/files/11406174/Archive.2.zip)

**Comment by Gauravprakash** [comment]
> I think most of the issues should be fixed with #2995.

but this should be fixed from library end

**Comment by Gauravprakash** [comment]
What's update here , please share your explanation for crashes occurring

**Comment by philipphofmann** [comment]
@Gauravprakash, could you please share the stacktrace of the crashes? That would help us to understand what's going on.

**Comment by armcknight** [comment]
Yes, the stack traces for each of these would help. I looked at the archive you uploaded, which had a stack trace for one of them, which is very helpful thanks! But the only other report in there was for a crash not listed in your screenshot, and involved no sentry code. If at all possible, please provide stack traces for the other crashes.

**Comment by armcknight** [comment]
@Gauravprakash we think we've fixed some of the crashes in #3018, based on the information in one of your stack traces and some other customer reports. Stay tuned for a patch release.

**Comment by armcknight** [comment]
@Gauravprakash We recently released 8.7.2 which should fix some crashes, please update and confirm if anything changes on your end.

**Comment by philipphofmann** [comment]
We think 8.7.2 should fix the issue, as pointed out in https://github.com/getsentry/sentry-cocoa/issues/2993#issuecomment-1551124145. Please reopen with more information otherwise.

**Comment by kahest** [comment]
@Gauravprakash the latest release is [8.7.3](https://github.com/getsentry/sentry-cocoa/releases/tag/8.7.3) - you can always check the [releases page](https://github.com/getsentry/sentry-cocoa/releases) to see what the latest release is.

**Comment by Gauravprakash** [comment]
[!] CocoaPods could not find compatible versions for pod "SentryPrivate":
  In Podfile:
    Sentry (from `https://github.com/getsentry/sentry-cocoa.git`, tag `8.7.3`) was resolved to 8.7.3, which depends on
      SentryPrivate (= 8.7.3)

what's expected here , could you please share pod dependency of sentry Private too

**Comment by kahest** [comment]
@brustolin ^ can you take a look?

**Comment by philipphofmann** [comment]
@Gauravprakash, please run `pod repo update`, which updates the spec repos located at ~/.cocoapods/repos in your home folder, and try again.

**Comment by brustolin** [comment]
SentryPrivate 8.7.3 seems to be in order
![image](https://github.com/getsentry/sentry-cocoa/assets/28265868/6ebecc93-8146-40d9-a908-2f798f358f94)

What @philipphofmann suggested will probably fix the issue.

**Comment by Gauravprakash** [comment]
Also can you please share pod file code, I need to check dependency script for 8.7.3

**Comment by brustolin** [comment]
> Also can you please share pod file code, I need to check dependency script for 8.7.3

https://github.com/getsentry/sentry-cocoa/blob/main/Sentry.podspec
https://github.com/getsentry/sentry-cocoa/blob/main/SentryPrivate.podspec

We dont have external dependency

**Comment by Gauravprakash** [comment]
@brustolin Regarding more information related to slowness , frozen frames :  I was checking your demo project for calculating frozen frames, slow frames and total frames using PrivateSentrySDKOnly , while using it seems it's a private Api which can be risky, as you may change or be removed in future versions of the SDK, causing compatibility issues. Can you please provide public API provided by the framework or library though which we can rcv data ?

**Comment by philipphofmann** [comment]
> Can you please provide public API provided by the framework or library though which we can rcv data ?

What exactly do you mean by that? Do you want to have an API for slow and frozen frames? If yes, can you please explain your use case for this?

**Comment by Gauravprakash** [comment]
like firebase performance we can track slow rendering and frozen frames though samples shared on console, similarly I can see Sentry helps to trace using PrivateSentrySDKOnly but it's not exposed publicly , so I want to ask Is it safe to use @philipphofmann as we need to collect data to send to our own server .
